cmake/modules/west.cmake

```diff
@@ -95,3 +95,55 @@ else()
     set(WEST WEST-NOTFOUND CACHE INTERNAL "West")
   endif()
 endif()
+
+# Verify blobs fetched using west. If the sha256 checksum isn't valid, a fatal
+# error message is printed.
+# Paths are relative to the module's zephyr/blobs directory.
+#
+# Usage:
+#   west_blobs_verify([MODULE module]
+#                     [PATHS path [paths...]])
+#
+# Example:
+# west_blobs_verify()                   # verify all blobs
+# west_blobs_verify(MODULE my_module)   # verify all blobs in "my_module"
+# west_blobs_verify(PATHS img/file.bin) # verify a single path
+function(west_blobs_verify)
+  if("${WEST}" STREQUAL "WEST-NOTFOUND")
+    message(WARNING "${CMAKE_CURRENT_FUNCTION} called without west, skipping")
+    return()
+  endif()
+
+  cmake_parse_arguments(WEST_BLOBS_VERIFY "" "MODULE" "PATHS" ${ARGN})
+
+  execute_process(
+    COMMAND ${WEST} blobs list ${WEST_BLOBS_VERIFY_MODULE} -f "{status} {path}"
+    OUTPUT_VARIABLE BLOBS_LIST_OUTPUT
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    COMMAND_ERROR_IS_FATAL ANY
+  )
+
+  string(REPLACE "\n" ";" BLOBS_LIST ${BLOBS_LIST_OUTPUT})
+
+  if(WEST_BLOBS_VERIFY_PATHS)
+    foreach(path ${WEST_BLOBS_VERIFY_PATHS})
+      message(STATUS "Verifying blob \"${path}\"")
+
+      # Each path that has a correct sha256 is prefixed with an A
+      if(NOT "A ${path}" IN_LIST BLOBS_LIST)
+        message(FATAL_ERROR "Blob for path \"${path}\" isn't valid. Update with: west blobs fetch")
+      endif()
+    endforeach()
+  else()
+    foreach(blob ${BLOBS_LIST})
+      separate_arguments(blob)
+      list(GET blob 0 status)
+      list(GET blob 1 path)
+      message(STATUS "Verifying blob \"${path}\"")
+
+      if(NOT "${status}" STREQUAL "A")
```

Comment:
Perhaps compare against `"M"` so that if a bin blob is missing it doesn't generate an error.
(at least for `binary_blobs_verify()`)

Reply:
A missing blob is treated the same way as an invalid, which makes more sense IMO.

Reply:
no, not in all cases.

we should not allow someone to introduce a `binary_blobs_verify()` in module `foo`, thus breaking the build if module `bar` is missing a blob.

A build should only breakif the specific blob required for this build is missing, hence the proposal for the `REQUIRED` together with `PATHS`.
It should not break because some blob, not used in this build, is missing.

Alternative remove the pure `binary_blobs_verify()` and always request `MODULE` or `PATHS`.

As example, if i'm build for infineon platform, the build should not fail because an espressif blob is missing.

Similar, if i'm build for Gecko SoC series `efr32mg12p`, then the build shouldn't fail because `librail_efr32xg13_gcc_release.a` is missing.

A build should only fail if the needed dependencies for that particular build is missing.

Reply:
I like the idea of a module to only allow verifying its own blobs, but for this to work, I need to get the current module name in `binary_blobs_verify`. Is that possible?

Reply:
we do have `module_name`, but currently considered build system internals.
If we have a use-case (like we do now), then we should create a corresponding `ZEPHYR_CURRENT_MODULE_NAME` here:
https://github.com/zephyrproject-rtos/zephyr/blob/982fc416f8df34977a720a372564b3b673e1fd1e/CMakeLists.txt#L575-L576

but before doing so, we must ensure that `west blobs` is actually using the Zephyr module's name, and not the `west project` name, as those two names may not be identical.